php-plugin/src/test/resources/checks/AvoidTryCatchFinallyCheck_NOK_FailsAllTryStatements.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+class SpecificException extends Exception
+{
+}
+
+function test()
+{
+    throw new SpecificException('Oopsie');
+}
+
+$file = 'file';
+try { // NOK {{Avoid using try-catch}}
+    $picture = PDF_open_image_file(
+        pdf_new(),
+        "jpeg",
+        $file,
+        "",
+        0
+    ); // This is the original statement, this works on PHP4
+} catch (Exception $ex) {
+    $msg = "Error opening $file";
+    throw new \Exception($msg);
+}
+
+try { // NOK {{Avoid using try-catch}}
+    throw new \Exception("Hello");
+} catch (\Exception $e) {
+    echo $e->getMessage() . " catch in\n";
+    throw $e;
+} finally {
+    echo $e->getMessage() . " finally \n";
+    throw new \Exception("Bye");
+}
+
+//FAILS with this RULE
```

Comment:
The rule fails with this use case?

Reply:
To be honnest, I don't really understand this rule (I mean, I do, but I don't totally see the point).

The test works properly if I write : 
```php
try { // NOK {{Avoid using try-catch}}
    throw new \Exception("Hello");
} catch (\Exception $e) {
    echo $e->getMessage() . " catch in\n";
    throw $e;
}
``` 
instead of : 
```php
try { 
    throw new \Exception("Hello");
} catch (\Exception $e) {
    echo $e->getMessage() . " catch in\n";
    throw $e;
}
```
because the purpose of this rule is to check if there are any usages of try catch statements.

Reply:
(I pushed some changes that made sens to me but I'm not totally sure if that's ok)

Reply:
Thank you for this explanation. I agree with you.
So I think we can discuss the revelance of this rule in another issue. I let you create it :+1:

Reply:
exaplanation of this rule : find EC34 in https://github.com/green-code-initiative/ecoCode/blob/main/RULES.md